pkg/symbolizer/symbolizer.go

```diff
@@ -31,170 +13,11 @@ type Frame struct {
 	Inline bool
 }
 
-type subprocess struct {
-	cmd     *exec.Cmd
-	stdin   io.Closer
-	stdout  io.Closer
-	input   *bufio.Writer
-	scanner *bufio.Scanner
-}
-
-func NewSymbolizer(target *targets.Target) *Symbolizer {
-	return &Symbolizer{target: target}
-}
-
-func (s *Symbolizer) Symbolize(bin string, pc uint64) ([]Frame, error) {
-	return s.SymbolizeArray(bin, []uint64{pc})
-}
-
-func (s *Symbolizer) SymbolizeArray(bin string, pcs []uint64) ([]Frame, error) {
-	sub, err := s.getSubprocess(bin)
-	if err != nil {
-		return nil, err
-	}
-	return symbolize(&s.interner, sub.input, sub.scanner, pcs)
-}
-
-func (s *Symbolizer) Close() {
-	for _, sub := range s.subprocs {
-		sub.stdin.Close()
-		sub.stdout.Close()
-		sub.cmd.Process.Kill()
-		sub.cmd.Wait()
-	}
-}
-
-func (s *Symbolizer) getSubprocess(bin string) (*subprocess, error) {
-	if sub := s.subprocs[bin]; sub != nil {
-		return sub, nil
-	}
-	addr2line, err := s.target.Addr2Line()
-	if err != nil {
-		return nil, err
-	}
-	cmd := osutil.Command(addr2line, "-afi", "-e", bin)
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		return nil, err
-	}
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		stdin.Close()
-		return nil, err
-	}
-	if err := cmd.Start(); err != nil {
-		stdin.Close()
-		stdout.Close()
-		return nil, err
-	}
-	sub := &subprocess{
-		cmd:     cmd,
-		stdin:   stdin,
-		stdout:  stdout,
-		input:   bufio.NewWriter(stdin),
-		scanner: bufio.NewScanner(stdout),
-	}
-	if s.subprocs == nil {
-		s.subprocs = make(map[string]*subprocess)
-	}
-	s.subprocs[bin] = sub
-	return sub, nil
-}
-
-func symbolize(interner *Interner, input *bufio.Writer, scanner *bufio.Scanner, pcs []uint64) ([]Frame, error) {
-	var frames []Frame
-	done := make(chan error, 1)
-	go func() {
-		var err error
-		defer func() {
-			done <- err
-		}()
-		if !scanner.Scan() {
-			if err = scanner.Err(); err == nil {
-				err = io.EOF
-			}
-			return
-		}
-		for range pcs {
-			var frames1 []Frame
-			frames1, err = parse(interner, scanner)
-			if err != nil {
-				return
-			}
-			frames = append(frames, frames1...)
-		}
-		for i := 0; i < 2; i++ {
-			scanner.Scan()
-		}
-	}()
-
-	for _, pc := range pcs {
-		if _, err := fmt.Fprintf(input, "0x%x\n", pc); err != nil {
-			return nil, err
-		}
-	}
-	// Write an invalid PC so that parse doesn't block reading input.
-	// We read out result for this PC at the end of the function.
-	if _, err := fmt.Fprintf(input, "0xffffffffffffffff\n"); err != nil {
-		return nil, err
-	}
-	input.Flush()
-
-	if err := <-done; err != nil {
-		return nil, err
-	}
-	return frames, nil
+type Symbolizer interface {
+	Symbolize(bin string, pcs ...uint64) ([]Frame, error)
```

Comment:
I wanted to make `bin` a Make param but it looks to intrusive for the `symbolizer.Cache`.